tests/auth_test.php

```diff
@@ -673,14 +673,11 @@ public function test_saml_login_complete_new_account(): void {
 
         // Checking that the events contain the expected values.
         $events = $sink->get_events();
-        $this->assertCount(4, $events);
+        $this->assertCount(2, $events);
```

Comment:
Now that user creation fully hydrates the user profile, the user creation and login flow only triggers two events, as the user updated event does not trigger because the code checks if fields have changed before doing the update flow.

Previously because only a minimal set of attributes was created, this check passed and the user update event fired.